action.yml

```diff
@@ -116,26 +116,32 @@ runs:
         export GITHUB_BASE_REF
         export GITHUB_HEAD_REF
         echo "::endgroup::"
+        set -euo pipefail
         bash ${{ github.action_path }}/src/scripts/fetch_through_merge_base.sh        
-    - id: fallback 
-      if: ${{ steps.fetch_through_merge_base.conclusion == 'failure' && inputs.fallback-fetch-all }}
+    # Note: `outcome` does not consider `continue-on-error`
+    # Note: fallback-fetch-all is a string, not a boolean, since type is not supported
+    # for inputs on composite actions (only workflows).
+    - id: check_for_failure
+      if: ${{ steps.fetch_through_merge_base.outcome == 'failure' && inputs.fallback-fetch-all == 'false' }}
       shell: bash
       env:
         WORKING_DIRECTORY: ${{ inputs.working-directory }}
       run: |
-        echo "::group::Falling back to fetching all...🚦"
-        cd $WORKING_DIRECTORY;
-        git fetch --all --unshallow;
-        python ${{ github.action_path }}/src/scripts/git_ungraft.py;
-        echo "::endgroup::"
-    - id: check_for_failure
-      if: ${{ steps.fetch_through_merge_base.conclusion == 'failure' }} 
+        echo "Action failed! ❌"
+        exit 1
+    - id: fallback 
+      if: ${{ steps.fetch_through_merge_base.outcome == 'failure' && inputs.fallback-fetch-all == 'true' }}
       shell: bash
       env:
+        BASE_REF: ${{ steps.fetch_through_merge_base.outputs.base-ref }}
+        HEAD_REF: ${{ steps.fetch_through_merge_base.outputs.head-ref }}
         WORKING_DIRECTORY: ${{ inputs.working-directory }}
       run: |
-        echo "Action failed! ❌"
-        exit 1
+        echo "::group::Falling back to fetching all...🚦"
+        cd $WORKING_DIRECTORY;
+        git fetch --all --unshallow --verbose;
+        python ${{ github.action_path }}/src/scripts/git_ungraft.py;
+        git merge-base $BASE_REF $HEAD_REF
```

Comment:
this is added so this will fail if `git fetch --all` doesn't work for some reason!